src/explore-education-statistics-common/src/services/util/combineMeasuresWithDuplicateLocationCodes.ts

```diff
@@ -86,126 +83,122 @@ export default function combineMeasuresWithDuplicateLocationCodes(
-                  name: combinedLocation.label,
-                  code,
-                },
-              },
-              measures: allAvailableMeasures.reduce(
-                (acc, measure) => ({
-                  ...acc,
-                  [measure]: '',
-                }),
-                {},
-              ),
-            } as TableDataResult;
-          });
-        },
-      );
-
-      const allAvailableLocationNames = Object.keys(
-        resultsGroupedByLocationName,
-      ).sort();
-
-      // Now for each combination of Time Period and Filters, produce a single combined row of data that merges
-      // the duplicate rows from each Location into one, using a strategy to merge a set of data for each measurement
-      // into a single value.
-      const deduplicatedResults = timePeriodFilterCombinations.flatMap(
-        combination => {
-          // For each measure, collect the value for that measure from each Location's results for this Time Period and
-          // Filter combination, or undefined if a Location does not have a value that matches this criteria.
-          const measureValuesForEachLocation = Object.keys(
-            combination.measures,
-          ).reduce((acc, measure) => {
-            const measureValues = allAvailableLocationNames.map(
-              locationName => {
-                const resultForLocationName = resultsGroupedByLocationName[
-                  locationName
-                ].find(
-                  result =>
-                    result.timePeriod === combination.timePeriod &&
-                    isEqual(combination.filters, result.filters),
-                );
-                return resultForLocationName?.measures[measure];
+        const allAvailableFilterCombinations = uniqWith(
+          rowsForTimePeriod.map(result => result.filters.sort()),
+          (a1, a2) => isEqual(a1, a2),
+        );
+        const allAvailableMeasures = uniq(
+          rowsForTimePeriod.flatMap(result => Object.keys(result.measures)),
+        );
+        return allAvailableFilterCombinations.flatMap(filters => {
+          return {
+            timePeriod,
+            filters,
+            geographicLevel: level,
+            location: {
+              [level]: {
+                name: combinedLocation.label,
+                code,
               },
-            );
-            return {
-              ...acc,
-              [measure]: measureValues,
-            };
-          }, {});
-
-          // Now for each measure, combine the values gathered from all of the Locations into a single value per measure.
-          const mergedMeasurements = mapValues(
-            measureValuesForEachLocation,
-            measurementsMergeStrategy,
+            },
+            measures: allAvailableMeasures.reduce(
+              (acc, measure) => ({
+                ...acc,
+                [measure]: '',
+              }),
+              {},
+            ),
+          } as TableDataResult;
```

Comment:
Looks like we don't need this cast anymore

Reply:
Removed, thanks